src/euler_one_qubit_decomposer.rs

```diff
@@ -31,7 +651,7 @@ fn complex_phase(x: Complex64) -> f64 {
 
 #[inline]
 fn mod_2pi(angle: f64) -> f64 {
-    (angle + PI) % (2. * PI) - PI
+    (angle + PI).rem_euclid(2. * PI) - PI
 }
```

Comment:
I was going to comment that `f64::rem_euclid` still isn't the same as Python's (`rem_euclid` is always non-negative, whereas Python's `%` matches the sign of the right-hand operator), but since we're only doing modulo a positive number, we're good.

Also, this function has lost the endpoint clamping that the Python version does.  Are we safe to do that?

Reply:
Yeah, this one was annoying to me too, I can leave a comment to this effect inline just for the historical record. Since it didn't matter in practice doing this was easier than the alternative which was to write our own equivalent of Python's `%` for rust which felt like a pain.

As for the clamping I thought it was safe to do since I didn't pass atol (I don't remember why I removed it, I think because nothing in the `Optimize1qGatesDecomposition` ever changed it from the default) into this function and just always used the default as `0`. So the clamping wouldn't really come up because we'd never be less than 0 here. But I might have been mistaken and can add it back if you think it's important (and if my memory of why I removed the atol parameter is correct I should add it back, probably in #9583 as setting the atol is part of that class's API).

Reply:
Yeah, agreed on `%` - we don't need to care about what happens with negative RHS because ours is a constant.

I'm happy to just add the clamping back in #9583 as appropriate - let's move to try and get this PR merged and onto the next one, where that part will actually be used.

Reply:
I added a comment on this here: https://github.com/Qiskit/qiskit-terra/pull/9578/commits/95f6974dbe3a6025d957ec61bd94e18674dcd95b